universum/modules/code_report_collector.py

```diff
@@ -39,29 +32,18 @@ def set_code_report_directory(self, project_root: str) -> None:
         if not os.path.exists(self.report_path):
             os.makedirs(self.report_path)
 
-    def prepare_environment(self, project_configs: List[ProjectConfig]) -> Variations:
-        afterall_steps: List[ProjectConfig] = []
-        for item in project_configs:
-            if not item.get("code_report", False):
+    def prepare_environment(self, project_config_variations: Variations) -> Variations:
+        afterall_steps: List[ProjectConfiguration] = []
```

Comment:
Would it be safe to use Variations here?

Reply:
in afterall_steps ?

Reply:
yes

Reply:
I am not sure I understand, we can either add Variations, or populate list of ProjectConfigurations and convert it at the end - there should be no difference really. I think current approach (convert in the end) has slightly lower overhead, since adding Variations has one more call to go through.

Reply:
It's about consistency (readability). I interpret Variations as a type used to store list of ProjectConfiguration instances, and the performance difference is minuscule.
BTW, probably we should also rename or alias Variations to something more meaningful.

Reply:
sure, I'll change it to Variations. I think there were several other places where that would make sense.
I propose following naming options: ConfigurationList, ConfigurationHolder.

Reply:
ConfigurationList looks good to me